sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -21,6 +21,9 @@ namespace Azure.Core.Serialization
     /// </summary>
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
     [JsonConverter(typeof(DynamicDataJsonConverter))]
+#if NET6_0_OR_GREATER
```

Comment:
@KrzysztofCwalina mentioned he had used this type with trimming, and it had somehow worked.  I'm curious if there's a reason that might be true in this case, and if it is, whether we would need this annotation.  I don't have a lot of context beyond this, but I'm interested in whether we could learn more about this.

Reply:
Using `JsonSerializer` without the source generator doesn't work with trimming. You will get warnings on these lines:

https://github.com/Azure/azure-sdk-for-net/blob/c1868b47146e4d1df68ff38ccb15c0a63447d1d0/sdk/core/Azure.Core/src/DynamicData/DynamicData.cs#L169-L171

https://github.com/Azure/azure-sdk-for-net/blob/c1868b47146e4d1df68ff38ccb15c0a63447d1d0/sdk/core/Azure.Core/src/DynamicData/DynamicData.cs#L200